### PR TITLE
Add weapon flow

### PR DESCRIPTION
**Pull request by Qnkacz**
Added post flow for weapons and underlying connected objects
solves #17

### CODE REVIEW

File: api/model/src/main/kotlin/wasik/api/restResource/items/model/IItem.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package wasik.api.restResource.items.model
+
+interface IItem {
```
Comment (tomasz-sieminski) [inline api/model/src/main/kotlin/wasik/api/restResource/items/model/IItem.kt]: Nitpick: Personally I prefer `ItemInterface`, depending on IDE, font etc `IItem` is hard to read

File: domain/logic/src/main/kotlin/wasik/items/mapper/DamageInfraMapper.kt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package wasik.items.mapper
+
+import model.damage.Damage
+import wasik.entity.DamageEntity
+
+interface DamageInfraMapper {
```
Comment (tomasz-sieminski) [inline domain/logic/src/main/kotlin/wasik/items/mapper/DamageInfraMapper.kt]: Question: Infrastructure or something else?

**Review by tomasz-sieminski — COMMENTED** [review]
Personally, I find this architecture confusing but if it works for you it's kay. Waiting for some unit tests for those mappers and I would really rethink this `InterfaceClass` and `ImplementationOfInterface` schema because later it will grow beyond your control and it will be hard to manage IMO

File: api/model/src/main/kotlin/wasik/api/restResource/items/model/IItem.kt (hunk 1 above)
Comment (Qnkacz) [reply api/model/src/main/kotlin/wasik/api/restResource/items/model/IItem.kt]: I wasn't sure how to approach it at first, so i went with the familiar .Net naming scheme but it doesn't work that well here.
Changed it

**Comment by Qnkacz** [comment]
I know that this architecture is sophisticated but it is like that by design and i will not change it.
The architecture isn't as hard, I've only added modules inside Api, domain and infra to separate the logic from the data classes as much as possible.
To help mitigate the problem, I've changed to packaging to better represent file placement inside the project. Hope that helps

**Pull request merged**